lib/Deserializer/functions.php

```diff
@@ -346,8 +348,22 @@ function functionCaller(Reader $reader, callable $func, string $namespace)
     }
 
     $funcArgs = [];
+    /**
+     * Even if $func gets "exploded" into an array, that array is still "callable"
+     * It should have elements that are the class and the function/method in the class.
+     */
     $func = is_string($func) && false !== strpos($func, '::') ? explode('::', $func) : $func;
-    $ref = is_array($func) ? new \ReflectionMethod($func[0], $func[1]) : new \ReflectionFunction($func);
+    if (is_callable($func)) {
+        if (is_array($func)) {
+            $ref = new \ReflectionMethod($func[0], $func[1]);
+        } elseif (($func instanceof \Closure) || is_string($func)) {
+            $ref = new \ReflectionFunction($func);
+        } else {
+            throw new \InvalidArgumentException(__METHOD__.' func parameter is not a callable array, string or closure.');
+        }
+    } else {
+        throw new \InvalidArgumentException(__METHOD__.' func parameter is not callable.');
```

Comment:
I can't see a way to get to this code.
When I try to make a unit test that passes a function reference like "SomeValidClass::unknownMethod", then PHP notices that during the call to `functionCaller`. Even though the string passed looks like a reasonable format for a callable thing, at run-time PHP checks if the method exists in the class. And the test fails with:
```
TypeError: Argument 2 passed to Sabre\Xml\Deserializer\functionCaller() must be callable, string given, called in /home/phil/git/sabre-io/xml/tests/Sabre/Xml/Deserializer/FunctionCallerTest.php on line 154
```

So PHP really verifies that the thing passed in as parameter `$func` not only looks like it could be callable, but that it is actually callable.

So in real-life we don't need all this double-checking.

I will see if there is a simpler way of coding this and keeping phpstan happy at the same time.